### PR TITLE
Ports "Simplifies and improves lone op counter"

### PR DESCRIPTION
**Pull request by ghost**
# Document the changes in your pull request

Ports "Simplifies and improves lone op counter" from beestation.

# Wiki Documentation
No
# Changelog

:cl:  [Rukofamicom](https://github.com/Rukofamicom)
tweak: simplified how lone op counter increments when nuclear disk is stationary
/:cl:

### CODE REVIEW

File: code/modules/antagonists/nukeop/equipment/nuclearbomb.dm [hunk 1]
```diff
@@ -669,7 +669,7 @@ This is here to make the tiles around the station mininuke change when it's arme
 		if(istype(loneop) && loneop.occurrences < loneop.max_occurrences && prob(loneop.weight))
 			loneop.weight = max(loneop.weight - 1, 0)
 			if(loneop.weight % 5 == 0 && SSticker.totalPlayers > 1)
-				message_admins("[src] is on the move (currently in [ADMIN_VERBOSEJMP(newturf)]). The weight of Lone Operative is now [loneop.weight].")
+			message_admins("[src] is on the move (currently in [ADMIN_VERBOSEJMP(newturf)]). The weight of Lone Operative is now [loneop.weight].")
```
Comment (ghost) [inline code/modules/antagonists/nukeop/equipment/nuclearbomb.dm]:
```suggestion
				message_admins("[src] is on the move (currently in [ADMIN_VERBOSEJMP(newturf)]). The weight of Lone Operative is now [loneop.weight].")
```

**Comment by ghost** [comment]
Fuck why does github so fucking hate me

**Comment by ynot01** [comment]
instead of `300 SECONDS` you could use `5 MINUTES`

**Comment by ghost** [comment]
Based

**Comment by JamieD1** [comment]
Can you please fix the compile, instead of shouting about why its broken, look at the reason in the compile errors:
image

File: code/modules/antagonists/nukeop/equipment/nuclearbomb.dm [hunk 2]
```diff
@@ -654,12 +654,12 @@ This is here to make the tiles around the station mininuke change when it's arme
 		CRASH("A fake nuke disk tried to call process(). Who the fuck and how the fuck")
 	var/turf/newturf = get_turf(src)
 	if(newturf && lastlocation == newturf)
-		if(last_disk_move < world.time - 5000 && prob((world.time - 5000 - last_disk_move)*0.0001))
+		if(last_disk_move < world.time - 300 SECONDS && world.time > 30 MINUTES)
```
Comment (ghost) [inline code/modules/antagonists/nukeop/equipment/nuclearbomb.dm]:
```suggestion
		if((last_disk_move < (world.time - 300 SECONDS)) && (world.time > 30 MINUTES))
```

**Comment by ghost** [comment]
> Can you please fix the compile, instead of shouting about why its broken, look at the reason in the compile errors: image

It is actually broken, it shows insteas of real existing errors non-existing ones

**Comment by JamieD1** [comment]
> > Can you please fix the compile, instead of shouting about why its broken, look at the reason in the compile errors: image
> 
> It is actually broken, it shows instead of real existing errors non-existing ones

I dont think thats the case. Considering it suddenly compiles now you fixed the error it displayed.

**Comment by JamieD1** [comment]
You literally made the lone op 5 times faster. Please can you not make changes without putting what you changed, yes you simplified it but you also made it add 5 to the weight every bloody tick.

**Comment by ghost** [comment]
> > > Can you please fix the compile, instead of shouting about why its broken, look at the reason in the compile errors: image
> > 
> > It is actually broken, it shows instead of real existing errors non-existing ones
> 
> I dont think thats the case. Considering it suddenly compiles now you fixed the error it displayed.

I fixed existing ones, not that what were displayed

**Review by Bibby0110 — CHANGES_REQUESTED** [review]
put the ported PR link

**Comment by ghost** [comment]
> put the ported PR link

okie

**Comment by JamieD1** [comment]
Should probably port the TG stuff instead which has more improvements

**Pull request closed without merging**